engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessManager.scala

```diff
@@ -16,7 +16,7 @@ trait ProcessManager {
   def findJobStatus(name: ProcessName) : Future[Option[ProcessState]]
 
   //TODO: this is very flink specific, how can we handle that differently?
-  def savepoint(name: ProcessName, savepointDir: String): Future[String]
+  def savepoint(name: ProcessName, savepointDir: Option[String], cancelProcess: Boolean): Future[String]
```

Comment:
What do you think about always making savepoint during cancel, and rename it to stop? Flink introduced second operation, stop: https://ci.apache.org/projects/flink/flink-docs-stable/ops/cli.html This concept will be close to their approach.

Proposed change will make this api more complex (three operations and two of them hidden in one method with flag).

Reply:
Done. I left savepointDir parameter because it can be useful when performing Flink upgrade to save savepoints temporarily in other directory.

Reply:
Indeed, can be useful :+1: